// @storyblok/react/dist/index.mjs
import { useState as f, useEffect as k } from "react";
import { registerStoryblokBridge as u } from "@storyblok/js";
import { BlockTypes as B, ComponentBlok as R, MarkTypes as C, TextTypes as E, apiPlugin as w, asTag as h, loadStoryblokBridge as A, registerStoryblokBridge as I, renderRichText as L, richTextResolver as F, storyblokEditable as P, useStoryblokBridge as _ } from "@storyblok/js";
import { storyblokInit as M } from "./core/init.mjs";
import { useStoryblokApi as m } from "./core/use-storyblok-api.mjs";
import { default as V } from "./core/storyblok-component.mjs";
import { useStoryblokState as $ } from "./core/use-storyblok-state.mjs";
import { useStoryblokRichText as c } from "./richtext.mjs";
import { default as q } from "./storyblok-rich-text.mjs";
import { convertAttributesInElement as D, isBridgeLoaded as G, isBrowser as H, isIframe as K, isServer as Q, isVisualEditor as U } from "./utils.mjs";
import { getComponent as X, getCustomFallbackComponent as Z, getEnableFallbackComponent as O, setComponents as oo } from "./core/state.mjs";
const g = (s, e = {}, o = {}) => {
  const [l, n] = f({}), i = typeof window < "u" && typeof window.storyblokRegisterEvent < "u", t = m();
  return k(() => {
    if (!t) {
      console.error(
        "You can't use useStoryblok if you're not loading apiPlugin."
      );
      return;
    }
    async function a() {
      const { data: r } = await t.get(
        `cdn/stories/${s}`,
        e
      );
      n(r.story), i && r.story.id && u(
        r.story.id,
        (y) => n(y),
        o
      );
    }
    a();
  }, [s, JSON.stringify(e), t]), t ? (o.resolveRelations = o.resolveRelations ?? e.resolve_relations, o.resolveLinks = o.resolveLinks ?? e.resolve_links, l) : null;
}, x = c;
export {
  B as BlockTypes,
  R as ComponentBlok,
  C as MarkTypes,
  V as StoryblokComponent,
  q as StoryblokRichText,
  E as TextTypes,
  w as apiPlugin,
  h as asTag,
  D as convertAttributesInElement,
  X as getComponent,
  Z as getCustomFallbackComponent,
  O as getEnableFallbackComponent,
  m as getStoryblokApi,
  G as isBridgeLoaded,
  H as isBrowser,
  K as isIframe,
  Q as isServer,
  U as isVisualEditor,
  A as loadStoryblokBridge,
  I as registerStoryblokBridge,
  L as renderRichText,
  F as richTextResolver,
  oo as setComponents,
  P as storyblokEditable,
  M as storyblokInit,
  g as useStoryblok,
  m as useStoryblokApi,
  _ as useStoryblokBridge,
  c as useStoryblokRichText,
  x as useStoryblokRichTextResolver,
  $ as useStoryblokState
};
