// @storyblok/react/dist/rsc.mjs
import { storyblokInit as r } from "./core/init.mjs";
import { useStoryblokServerRichText as l } from "./server/richtext.mjs";
import { default as p } from "./server/server-component.mjs";
import { default as k } from "./server/server-story.mjs";
import { default as b } from "./server/server-storyblok-richtext-component.mjs";
import { default as m } from "./rsc/live-editing.mjs";
import { default as S } from "./rsc/story.mjs";
import { BlockTypes as x, ComponentBlok as d, MarkTypes as g, TextTypes as u, apiPlugin as T, asTag as c, loadStoryblokBridge as C, registerStoryblokBridge as v, renderRichText as B, richTextResolver as h, storyblokEditable as R, useStoryblokBridge as E } from "@storyblok/js";
import { getComponent as F, getCustomFallbackComponent as I, getEnableFallbackComponent as L, setComponents as M } from "./core/state.mjs";
import { useStoryblokApi as j } from "./core/use-storyblok-api.mjs";
export {
  x as BlockTypes,
  d as ComponentBlok,
  g as MarkTypes,
  m as StoryblokLiveEditing,
  p as StoryblokServerComponent,
  b as StoryblokServerRichText,
  k as StoryblokServerStory,
  S as StoryblokStory,
  u as TextTypes,
  T as apiPlugin,
  c as asTag,
  F as getComponent,
  I as getCustomFallbackComponent,
  L as getEnableFallbackComponent,
  j as getStoryblokApi,
  C as loadStoryblokBridge,
  v as registerStoryblokBridge,
  B as renderRichText,
  h as richTextResolver,
  M as setComponents,
  R as storyblokEditable,
  r as storyblokInit,
  E as useStoryblokBridge,
  l as useStoryblokServerRichText
};
