// @storyblok/react/dist/ssr.mjs
import { storyblokInit as r } from "./core/init.mjs";
import { useStoryblokApi as l, useStoryblokApi as p } from "./core/use-storyblok-api.mjs";
import { default as s } from "./core/storyblok-component.mjs";
import { useStoryblokState as b } from "./core/use-storyblok-state.mjs";
import { BlockTypes as n, ComponentBlok as y, MarkTypes as S, TextTypes as i, apiPlugin as x, asTag as f, loadStoryblokBridge as u, registerStoryblokBridge as d, renderRichText as g, richTextResolver as T, storyblokEditable as C, useStoryblokBridge as c } from "@storyblok/js";
import { useStoryblokServerRichText as B } from "./server/richtext.mjs";
import { default as R } from "./server/server-component.mjs";
import { default as E } from "./server/server-story.mjs";
import { default as I } from "./server/server-storyblok-richtext-component.mjs";
import { getComponent as P, getCustomFallbackComponent as j, getEnableFallbackComponent as q, setComponents as w } from "./core/state.mjs";
export {
  n as BlockTypes,
  y as ComponentBlok,
  S as MarkTypes,
  s as StoryblokComponent,
  R as StoryblokServerComponent,
  I as StoryblokServerRichText,
  E as StoryblokServerStory,
  i as TextTypes,
  x as apiPlugin,
  f as asTag,
  P as getComponent,
  j as getCustomFallbackComponent,
  q as getEnableFallbackComponent,
  l as getStoryblokApi,
  u as loadStoryblokBridge,
  d as registerStoryblokBridge,
  g as renderRichText,
  T as richTextResolver,
  w as setComponents,
  C as storyblokEditable,
  r as storyblokInit,
  p as useStoryblokApi,
  c as useStoryblokBridge,
  B as useStoryblokServerRichText,
  b as useStoryblokState
};
